test/integration/fixtures/test_dynamic.html

```diff
@@ -0,0 +1,15 @@
+<html>
+
+<body>
+  <p id="print-ready" ready-to-print="false"></p>
+
+  <script>
+    setTimeout(() => {
+      const element = document.getElementById('print-ready');
+      element.innerText = "Dynamic content from Javascript";
+      element.setAttribute('ready-to-print', true);
+    }, 1000);
```

Comment:
I had a tingling thought about that and yes, it is a risk: definitely worth mentioning in the docs.

It could be mitigated by getting the attributes of the element first, just to verify that we can continue if the attribute is there. It would not hurt to add that to the protocol. AFAIK the developer protocol only sends events on elements that the client is aware.

A test could have static HTML page with the attribute set and it should pass.

Reply:
🤔  suggestion: focus on getting this PR ready first (maybe mention the race condition with a one-liner in the docs), and find a solution afterwards? Adding runtime conditionals to the protocol will be a bit messy, I think.

Reply:
Thinking further I don't think my mitigation idea would work because of the missing "runtime conditionals" you mention: there is no way to branch in the protocol after checking the value.